lib/AST/ASTDumper.cpp

```diff
@@ -1978,7 +1978,8 @@ namespace {
 
       printFlag(!ABIRoleInfo(D).providesAPI(), "abi_only");
 
-      printSourceRange(D->getSourceRange(), &D->getASTContext());
+      if (D->getStartLoc().isValid() && D->getEndLoc().isValid())
```

Comment:
```suggestion
      if (D->getSourceRange().isValid())
```
If you did this because there's a case `D->getStartLoc().isValid() != D->getEndLoc().isValid()`, we should fix that.

Reply:
Okay, I can drop that change.